traits/trait_converters.py

```diff
@@ -42,6 +42,33 @@ def trait_cast(obj):
     return None
 
 
+def as_ctrait(obj):
+    """ Convert to CTrait if the object knows how, else raise TraitError.
+
+    Parameters
+    ----------
+    obj: An object that supports conversion to `CTrait`. Refer
+        documentation for when the object supports this conversion.
+
+    Returns
+    -------
+    ctrait.CTrait
+
+    Raises
+    ------
+    TypeError if the object does not support conversion to CTrait.
+    """
+    if isinstance(obj, type) and hasattr(obj, 'instantiate_and_get_ctrait'):
+        ctrait = obj.instantiate_and_get_ctrait()
+    elif not isinstance(obj, type) and hasattr(obj, 'as_ctrait'):
+        ctrait = obj.as_ctrait()
+    else:
+        raise TypeError(
+            "Object {!r} does not support conversion to CTrait".format(obj))
+
+    return ctrait
```

Comment:
Code style nit: you could also consider returning immediately in the first two branches, which would make the code a bit simpler. It's fine as it is, though; I leave it up to you. (We don't generally subscribe to the "functions should only have a single return" school of thought ...)

Reply:
I fixed the code style nit and docstring formatting.